### PR TITLE
Chapter 13: User microposts

### PR DESCRIPTION
**Pull request by stephanie1997-kek**

### CODE REVIEW

**Comment by stephanie1997-kek** [comment]
ready

File: app/assets/javascripts/microposts.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (saobv-2223) [inline app/assets/javascripts/microposts.coffee]: remove file that never used

File: app/assets/stylesheets/microposts.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Microposts controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/microposts.scss]: remove file that never used

File: app/controllers/microposts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  def create
+  	@micropost = current_user.microposts.build micropost_params
+  	@micropost.image.attach micropost_params[:image]
+  	if @micropost.save
+  	  flash[:success] = "Micropost created!"
+  	  redirect_to root_url
+  	else
+  	  @feed_items = current_user.feed.page(params[:page]).per Settings.page
+  	  render 'static_pages/home'
+  	end
```
Comment (saobv-2223) [inline app/controllers/microposts_controller.rb]: indenting only 2 spaces

**Review by saobv-2223 — COMMENTED** [review]
check indenting again.

**Comment by stephanie1997-kek** [comment]
ready

File: app/controllers/microposts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  def create
+    @micropost = current_user.microposts.build micropost_params
+    @micropost.image.attach micropost_params[:image]
+    if @micropost.save
+  	 flash[:success] = t ".success_mess_created"
+  	 redirect_to root_path
+    else
+      @feed_items = current_user.feed.page(params[:page]).per Settings.collections.microposts_default_page
+      flash.now[:danger] = t ".fail_mess_created"
+      render 'static_pages/home'
```
Comment (saobv-2223) [inline app/controllers/microposts_controller.rb]: using "

File: app/controllers/microposts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    end
+  end
+
+  def destroy
+    @micropost.destroy
+    flash[:success] = t ".success_mess_deleted"
+    else
+      flash[:warning] = t ".fail_mess_deleted"
+    end
```
Comment (saobv-2223) [inline app/controllers/microposts_controller.rb]: add if

**Comment by stephanie1997-kek** [comment]
ready

File: app/helpers/microposts_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module MicropostsHelper
+end
```
Comment (saobv-2223) [inline app/helpers/microposts_helper.rb]: remove file that never used

File: app/javascript/packs/uploadImage.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+$(function () {
+  $('#micropost_image').bind('change', function() {
+    var size_in_megabytes = this.files[0].size/1024/1024;
+    if (size_in_megabytes > 5) {
+      alert(I18n.t("shared.micropost_form.alert_mess"));
```
Comment (saobv-2223) [inline app/javascript/packs/uploadImage.js:5]:
```suggestion
      alert(I18n.t("shared.micropost_form.alert_mess"));
      $(this).val('');
```

File: app/models/micropost.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+  belongs_to :user
+  has_one_attached :image
+  validates :user_id, presence: true
+  validates :content, presence: true, length: { maximum: Settings.validations.content.max_length }
+  validates :image, content_type: { in: Settings.image.type.split, message: I18n.t("errors.messages.invalid_image_format") }, size: { less_than: Settings.image.size.megabytes, message: I18n.t("errors.messages.invalid_image_size") }
+  scope :recent_posts, -> { order created_at: :desc }
+
+  def display_image
+    image.variant resize_to_limit: [Settings.image.resize, Settings.image.resize]
+  end
```
Comment (saobv-2223) [inline app/models/micropost.rb]:
```suggestion
  belongs_to :user
  has_one_attached :image
  
  validates :user_id, presence: true
  validates :content, presence: true, length: { maximum: Settings.validations.content.max_length }
  validates :image, content_type: { in: Settings.image.type.split, message: I18n.t("errors.messages.invalid_image_format") }, size: { less_than: Settings.image.size.megabytes, message: I18n.t("errors.messages.invalid_image_size") }
  
  scope :recent_posts, -> { order created_at: :desc }

  def display_image
    image.variant resize_to_limit: [Settings.image.resize, Settings.image.resize]
  end
```

**Comment by stephanie1997-kek** [comment]
ready

File: app/views/static_pages/home.html.erb [hunk 9]
```diff
@@ -1,15 +1,32 @@
-<% provide :title, t(".home_title") %>
+<% if logged_in? %>
+  <div class="row">
+    <% provide :title, t(".home_title") %>
+    <aside class="col-md-4">
+      <section class="user_info">
+  	    <%= render 'shared/user_info' %>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]:
```suggestion
  	  <%= render "shared/user_info" %>
```

File: app/views/static_pages/home.html.erb [hunk 10]
```diff
@@ -1,15 +1,32 @@
...
+      </section>
+      <section class="micropost_form">
+        <%= render 'shared/micropost_form' %>
+      </section>
+    </aside>
+    <div class="col-md-8">
+      <h3><%= t ".micropost_heading" %></h3>
+  	  <%= render "shared/feed" %>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]:
```suggestion
  	<%= render "shared/feed" %>
```

File: app/views/static_pages/home.html.erb [hunk 11]
```diff
@@ -1,15 +1,32 @@
-<% provide :title, t(".home_title") %>
+<% if logged_in? %>
+  <div class="row">
+    <% provide :title, t(".home_title") %>
+    <aside class="col-md-4">
+      <section class="user_info">
+  	    <%= render 'shared/user_info' %>
+      </section>
+      <section class="micropost_form">
+        <%= render 'shared/micropost_form' %>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]:
```suggestion
        <%= render "shared/micropost_form" %>
```

**Comment by stephanie1997-kek** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**